src/overlay/Peer.cpp

```diff
@@ -1868,7 +1991,103 @@ Peer::PeerMetrics::PeerMetrics(VirtualClock::time_point connectedTime)
     , mDuplicateFloodMessageRecv(0)
     , mUniqueFetchMessageRecv(0)
     , mDuplicateFetchMessageRecv(0)
+    , mTxHashReceived(0)
     , mConnectedTime(connectedTime)
 {
 }
+
+bool
+Peer::supportsAdverts() const
+{
+    return ((mApp.getConfig().OVERLAY_PROTOCOL_VERSION >=
+             MIN_OVERLAY_VERSION_FOR_FLOOD_ADVERT) &&
+            (mRemoteOverlayVersion >= MIN_OVERLAY_VERSION_FOR_FLOOD_ADVERT));
+}
+
+void
+Peer::advertizeTransaction(Hash const& txHash)
+{
+    // TODO: Remove this before production.
+    // This leads to way too many messages.
+    // But this is really useful for debugging.
+    CLOG_TRACE(Overlay, "Peer::advertizeTransaction -- queueing {} for {}",
+               hexAbbrev(txHash), mApp.getConfig().toShortString(getPeerID()));
+    mPendingAdvertMsg.floodAdvert().txHashes.emplace_back(txHash);
+}
+
+void
+Peer::sendTxDemand(std::vector<Hash> const& demands)
+{
+    if (demands.size() > 0)
+    {
+        auto msg = std::make_shared<StellarMessage>();
+        msg->type(FLOOD_DEMAND);
+        for (auto const& demand : demands)
+        {
+            msg->floodDemand().txHashes.push_back(demand);
+        }
+        std::weak_ptr<Peer> weak(static_pointer_cast<Peer>(shared_from_this()));
+        CLOG_TRACE(Overlay, "Peer::sendTxDemand -- demanding {} txns from {}",
+                   demands.size(), mApp.getConfig().toShortString(getPeerID()));
+        mApp.postOnMainThread(
+            [weak, msg = std::move(msg)]() {
+                auto strong = weak.lock();
+                if (strong)
+                {
+                    strong->sendMessage(msg);
+                }
+            },
+            "sendTxDemand");
+        ++mPeerMetrics.mTxDemandSent;
+    }
+}
+
+void
+Peer::flushAdvert()
+{
+    auto& txHashes = mPendingAdvertMsg.floodAdvert().txHashes;
+
+    if (txHashes.size() > 0)
+    {
+        std::weak_ptr<Peer> weak(static_pointer_cast<Peer>(shared_from_this()));
+        auto msg = std::make_shared<StellarMessage>(mPendingAdvertMsg);
+        CLOG_TRACE(Overlay, "Peer::flushAdvert -- flushing {} tx hashes to {}",
+                   txHashes.size(),
+                   mApp.getConfig().toShortString(getPeerID()));
+        mApp.postOnMainThread(
```

Comment:
It was because Graydon's PR did it this way. I just looked at other parts of `Peer.cpp` and it looks like they just call `sendMessage` directly. Would that be appropriate here?